packages/mobile/src/navigator/DrawerNavigator.tsx

```diff
@@ -216,6 +229,9 @@ function CustomDrawerContent(props: DrawerContentComponentProps<DrawerContentOpt
 export default function DrawerNavigator() {
   const { t } = useTranslation(Namespaces.global)
   const isCeloEducationComplete = useSelector((state) => state.goldToken.educationCompleted)
+  const name = useSelector(nameSelector)
+  const dispatch = useDispatch()
+  const inviteLink = useAsync(generateInviteLink, []).result
```

Comment:
Maybe, yeah, I did it this way because I need the `inviteLink` to create the message to share and that is needed to toggle the modal, but there's probably a better way to do that, I'll give it some more thought

Reply:
Changed the place of the generation of the link so it happens when the invite button is tapped :)